### DISCUSSION TITLE
git, display the current branch

### DISCUSSION THREAD

**Discussion by antyang26** [intersystems/git-source-control#81] [Q&A] [answered]
my book info:     MacBook Pro (13-inch, M1, 2020) /    Apple M1

1. when I checkout branch, I don't konw  the branch
2. like this
![image](https://user-images.githubusercontent.com/28669095/143013166-8509b52a-d41f-48b0-afd7-656926a1b953.png)

3. so ,I search the net ,somebody do like this  in .bash_profile
function git_branch {
  branch="`git branch 2>/dev/null | grep "^*" | sed -e "s/^* //"`"
  if [ "${branch}" != "" ];then
    if [ "${branch}" = "(no branch)" ];then
      branch="(`git rev-parse --short HEAD`...)"
    fi
    echo " ($branch)"
  fi
}
export PS1='u@h [33[01;36m]W[33[01;32m]$(git_branch)[33[00m] $ '

4. I do ,then I run    source .bash_profile  to restart it
  and get the result like this 
![image](https://user-images.githubusercontent.com/28669095/143013547-7c32d3b5-9ece-477d-b519-c4b2e0ad07b4.png)

5. and the file don't restart ,  what should I do?

**Comment by isc-tleavitt** [comment, marked as answer]
@antyang26, I think one possible issue is the use of single quotes (rather than double quotes) on the export PS1... line at the end. If changing that doesn't fix it, I'd try another solution like https://coderwall.com/p/fasnya/add-git-branch-name-to-bash-prompt .

Also, generally, there are probably better places than this to get answers to generic git questions.
